chef/cookbooks/neutron/recipes/l3.rb

```diff
@@ -53,16 +53,31 @@
 end
 
 
-# Enable ip forwarding on network node
+# Enable ip forwarding on network node for SLE11
 ruby_block "edit /etc/sysconfig/sysctl for IP_FORWARD" do
   block do
     rc = Chef::Util::FileEdit.new("/etc/sysconfig/sysctl")
     rc.search_file_replace_line(/^IP_FORWARD=/, 'IP_FORWARD="yes"')
     rc.write_file
   end
-  only_if { node[:platform] == "suse" }
+  only_if { node[:platform] == "suse" && node[:platform_version].to_f < 12.0 }
 end
 
+# Enable ip forwarding on network node for SLE12
+ruby_block "edit /etc/sysctl.d/99-sysctl.conf for net.ipv4.ip_forward" do
+  block do
+    rc = Chef::Util::FileEdit.new("/etc/sysctl.d/99-sysctl.conf")
+    rc.search_file_replace_line(/^net.ipv4.ip_forward =/, 'net.ipv4.ip_forward = 1')
+    rc.write_file
+  end
+  only_if { node[:platform] == "suse" && node[:platform_version].to_f >= 12.0 }
+end
```

Comment:
I guess you need to put this code after `directory "create /etc/sysctl.d for enable-ip_forward" do`, or is the directory just always existing?